### ISSUE TITLE
fix CI: /var/lib/acme is not correctly cached on ns.testrun.org

### ISSUE THREAD

**Issue by missytake** [deltachat/chatmail#302]
CI fails currently, because it always tries to request new certs, but runs into the Let's Encrypt rate limit.

My guess is that /var/lib/acme is not correctly cached on ns.testrun.org. On both machines, they look like this right now:

```
/var/lib/acme
├── accounts
│   └── acme-v02.api.letsencrypt.org%2fdirectory
│       ├── 2aq44le2l75gv5t6iuv4zshhvv54nl57uh7aud5ug7nv2e7zsq5a
│       │   └── privkey
│       ├── 3a7y5cdcokldckxchcavh3ayi34gqi3l2etr47clmazuuxlebesq
│       │   └── privkey
│       ├── 433uh75o3b2rbzbp6v5f4n7kqljm22mqvfra424is7iltlz2jf4q
│       │   └── privkey
│       └── jo2alzaczy742a7dhzncbcv2wo2ldrihbfpshvabfe2j75d3pwtq
│           └── privkey
├── acme
│   ├── accounts
│   │   └── acme-v02.api.letsencrypt.org%2fdirectory
│   │       ├── 3a7y5cdcokldckxchcavh3ayi34gqi3l2etr47clmazuuxlebesq
│   │       │   └── privkey
│   │       └── 433uh75o3b2rbzbp6v5f4n7kqljm22mqvfra424is7iltlz2jf4q
│   │           └── privkey
│   ├── acme
│   │   ├── accounts
│   │   │   └── acme-v02.api.letsencrypt.org%2fdirectory
│   │   │       └── 433uh75o3b2rbzbp6v5f4n7kqljm22mqvfra424is7iltlz2jf4q
│   │   │           └── privkey
│   │   ├── certs
│   │   │   ├── lw2gx2tow3oh6bulj5dxxhsblj4pdvydcbrq2hiwanm2ygioof6q
│   │   │   │   ├── account -> ../../accounts/acme-v02.api.letsencrypt.org%2fdirectory/433uh75o3b2rbzbp6v5f4n7kqljm22mqvfra424is7iltlz2jf4q
│   │   │   │   ├── cert
│   │   │   │   ├── chain
│   │   │   │   ├── fullchain
│   │   │   │   ├── privkey -> ../../keys/exe5zlx5qdpukci6ent7lu53mxzrlcb3rhzqxf3jpb3sv2enzova/privkey
│   │   │   │   └── url
[...]
```

But only 3 levels deep, so maybe it's fine and it's actually something else...

**Comment by hagenest** [comment]
I'm struggling to find the problem atm. I suppose I could set up a rsyslog-server or something with grafana, so that the logs on staging are persistent?

**Comment by missytake** [comment]
Let's revert https://github.com/deltachat/chatmail/commit/127d9d64606cde886ae6e1010c7f2ca79a98cc63 first, and see whether it works now?

**Comment by missytake** [comment]
maybe the problem was completely unrelated.

**Comment by hagenest** [comment]
I guess we could change the condition introduced in https://github.com/deltachat/chatmail/commit/127d9d64606cde886ae6e1010c7f2ca79a98cc63 to only request new certs if they are 2.9+ months old. But that'd also only be a bandage, I wonder how the problem happened in the first place.

**Comment by hagenest** [comment]
I'm confused. Reverting https://github.com/deltachat/chatmail/commit/127d9d64606cde886ae6e1010c7f2ca79a98cc63 runs into the rate limit again, but why? There shouldn't have been a request in the last 168 at all...

See: https://github.com/deltachat/chatmail/actions/runs/9260733717/job/25475017517?pr=311

**Comment by missytake** [comment]
> only request new certs if they are 2.9+ months old

acmetool generally doesn't initiate an ACME procedure if there is still a relatively fresh cert. So the problem can only appear because there is no fresh cert / no cert at all.

In our current case with the cert that expires on June 15, there is no fresh cert.

**Comment by hagenest** [comment]
> > I'm confused. Reverting [127d9d6](https://github.com/deltachat/chatmail/commit/127d9d64606cde886ae6e1010c7f2ca79a98cc63) runs into the rate limit again, but why? There shouldn't have been a request in the last 168h at all...
> 
> Hm, it says `retry after 2024-05-29T06:54:10`, let's just follow that advice.

yeah, now it says retry after 2024-06-02. Odd. Are we 100% sure that we don't have anything else that requests certs for staging.testrun.org?
